src/commands/sarif/validation.ts

```diff
@@ -0,0 +1,78 @@
+import fs from 'fs'
+
+
+import type {ErrorObject} from 'ajv'
+
+import Ajv from 'ajv'
+import addFormats from 'ajv-formats'
+
+import sarifJsonSchema from './json-schema/sarif-schema-2.1.0.json'
+
+/**
+ * Validate the SARIF file against the SARIF schema.
+ *
+ * @param sarifReportPath - the path of the SARIF file
+ */
+export const validateSarif = (sarifReportPath: string) => {
+  const ajv = new Ajv({allErrors: true})
+  addFormats(ajv)
+  const sarifJsonSchemaValidate = ajv.compile(sarifJsonSchema)
+  try {
+    const sarifReportContent = JSON.parse(String(fs.readFileSync(sarifReportPath)))
+    const valid = sarifJsonSchemaValidate(sarifReportContent)
+    if (!valid) {
+      const errors = sarifJsonSchemaValidate.errors || []
+      const errorMessages = errors.map((error: ErrorObject) => {
+        return `${error.instancePath}: ${error.message}`
+      })
+
+      return errorMessages.join('\n')
+    }
+  } catch (error) {
+    return error.message
+  }
+
+  return undefined
+}
+
+/**
+ * Functions that looks for errors specific to how Datadog processes SARIF file. This way, we
+ * show the error directly to the user instead of uploading a file we cannot process in our backend.
+ *
+ * The function returns undefined if everything is good, otherwise, it returns the error to show.
```

Comment:
We should support multiple errors not just one

Reply:
Why? The validation works the same way. We should keep it simple.